gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -184,33 +195,22 @@ public long renewToken(final String token, long renewInterval) throws UnknownTok
   @Override
   public void revokeToken(final JWTToken token) throws UnknownTokenException {
     if (token == null) {
-      throw new IllegalArgumentException("Token data cannot be null.");
+      throw new IllegalArgumentException("Token identifier cannot be null.");
     }
 
-    revokeToken(token.getPayload());
+    revokeToken(TokenUtils.getTokenId(token));
   }
 
   @Override
-  public void revokeToken(final String token) throws UnknownTokenException {
+  public void revokeToken(final String tokenId) throws UnknownTokenException {
     /* no reason to keep revoked tokens around */
-    removeToken(token);
-    log.revokedToken(TokenUtils.getTokenDisplayText(token));
+    removeToken(tokenId);
+    log.revokedToken(tokenId);
   }
 
   @Override
   public boolean isExpired(final JWTToken token) throws UnknownTokenException {
-    return isExpired(token.getPayload());
-  }
-
-  @Override
-  public boolean isExpired(final String token) throws UnknownTokenException {
-    boolean isExpired;
-    isExpired = isUnknown(token); // Check if the token exist
-    if (!isExpired) {
-      // If it not unknown, check its expiration
-      isExpired = (getTokenExpiration(token) <= System.currentTimeMillis());
-    }
-    return isExpired;
+    return getTokenExpiration(token) <= System.currentTimeMillis();
```

Comment:
This method would throw `UnknownTokenException` if a token is not found. Exceptions are expensive (filling up stack trace and unwinding it every time it is passed to a new method) so should `getTokenExpiration(token)` return some negative value instead?

Reply:
Since I think the UnknownTokenException scenario should be exceptional, I think it's ok as it is. It's not intended to be a regularly occurring logic path. Does that make sense?

Reply:
Yup, it does.